cpp/include/filling.hpp

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf.h>
+
+namespace cudf {
+
+void fill(gdf_column *column, gdf_scalar const& value, 
```

Comment:
Almost every other libcudf API is out-of-place. We should think about what the implications are for having a mix of in-place and out-of-place functions.

Reply:
Agree. But every time I thought about how a `fill()` would be used, I couldn't think of a case where the user wouldn't want to write directly to the column.

Reply:
I think we should provide the in-place option if and when doing the operation in-place can have significant savings in memory and/or performance. See https://github.com/rapidsai/cudf/issues/1468#issuecomment-498490120